packages/flutter/lib/src/material/segmented_button.dart

```diff
@@ -552,6 +552,7 @@ class SegmentedButtonState<T> extends State<SegmentedButton<T>> {
 
       return MergeSemantics(
         child: Semantics(
+          selected: segmentSelected,
           checked: segmentSelected,
```

Comment:
@chunhtai WDYT of selected versus checked here?

Reply:
selected will assigned https://developer.apple.com/documentation/uikit/uiaccessibilitytraits/selected?language=objc

which makes sense to me for segmented button.

but all these selected/checked/toggled bothered me a lot since different platform uses where it has these state to decide what component this semantics node present.

At some point I think we should use a semantics role with only one on and off state to replace all these selected/checked/toggled states. Util then, this is fine for now